build fails when /bin/sh points to dash

If system shell is dash (`/bin/sh -> dash`) build fails like this:
```
for var in GETTEXT_PACKAGE PACKAGE_VERSION PACKAGE localedir libdir libexecdir datadir sysconfdir pythondir pkgpythondir pkgdatadir pkgconfdir pkggladedir pkglibexecdir LIBSMBIOS_C_SONAME clidir __VERSION__  ;\
do      \
        perl -p -i -e "s|^$var\s*=.*|$var=\"${!var}\"|"  src/python/_vars.py;\
done
/bin/dash: 3: Bad substitution
make: *** [Makefile:3147: src/python/_vars.py] Error 2
```
because dash doesn't know about `${!var}` expansion.
Please make sure to use a dash with builtin LINENO support or else build system silently switches to bash which obviously won't trigger this issue.

This has also been reported downstream at https://bugs.gentoo.org/715202

--- a/Makefile-std
+++ b/Makefile-std
@@ -38,7 +38,7 @@ install-data-hook:
 	    file=$(DESTDIR)/$$i                     ;\
 	    for var in $(REPLACE_VARS)	;\
 	    do	\
-	        perl -p -i -e "s|^$$var\s*=.*|$$var=\"$${!var}\"|"  $$file;\
+	        var="$$var" perl -p -i -e 's|^\Q$$ENV{var}\E\s*=.*|$$ENV{var}="$$ENV{$$ENV{var}}"|'  $$file;\
 	    done	;\
 	done
 
@@ -48,7 +48,7 @@ install-exec-hook:
 	    file=$(DESTDIR)/$$i                     ;\
 	    for var in $(REPLACE_VARS)	;\
 	    do	\
-	        perl -p -i -e "s|^$$var\s*=.*|$$var=\"$${!var}\"|"  $$file;\
+	        var="$$var" perl -p -i -e 's|^\Q$$ENV{var}\E\s*=.*|$$ENV{var}="$$ENV{$$ENV{var}}"|'  $$file;\
 	    done	;\
 	done
 

--- a/src/python/Makefile.am
+++ b/src/python/Makefile.am
@@ -29,5 +29,5 @@ src/python/_vars.py: src/python/libsmbios_c/_vars.py  configure Makefile config.
 	cp $< $@
 	for var in $(REPLACE_VARS)	;\
 	do	\
-		perl -p -i -e "s|^$$var\s*=.*|$$var=\"$${!var}\"|"  $@;\
+		var="$$var" perl -p -i -e 's|^\Q$$ENV{var}\E\s*=.*|$$ENV{var}="$$ENV{$$ENV{var}}"|'  $@;\
 	done

